status-reporter/status_reporter/rabbitmq_reporter.py

```diff
@@ -13,17 +13,110 @@
+        self._update_node_status(services, statuses, config)
+        self._update_cluster_status(services, statuses, config)
+        status = get_node_status(statuses)
+        return status, services
+
+    @staticmethod
+    def _rabbitmq_service_not_running(statuses):
+        return statuses[0] == NodeServiceStatus.INACTIVE
+
+    def _update_cluster_status(self, services, statuses, config):
+        cluster_status, cluster_extra_info = self._get_cluster_status(config)
+        services[RABBITMQ_SERVICE_KEY]['extra_info']['cluster_status'] = \
+            cluster_extra_info
+        statuses.append(cluster_status)
+
+    def _get_cluster_status(self, config):
+        cluster_status_info = self._rabbitmqctl_cluster_status(config)
+        if not cluster_status_info:
+            return NodeServiceStatus.INACTIVE, {}
+
+        return NodeServiceStatus.ACTIVE, cluster_status_info
+
+    def _rabbitmqctl_cluster_status(self, config):
+        cmd = self._get_cluster_status_cmd(config)
+        runner = LocalCommandRunner()
+        try:
+            cluster_status_info = json.loads(runner.run(cmd).std_out)
+        except CommandExecutionException as error:
+            logger.error(
+                'Failed getting RabbitMQ cluster-status due to '
+                '{0}'.format(error))
+            return None
+
+        return cluster_status_info
+
+    @staticmethod
+    def _get_cluster_status_cmd(config):
+        longnames = '--longnames' if config['use_long_name'] else ''
+        nodename = config['nodename']
+        return ('sudo rabbitmqctl -n {nodename} {longnames} cluster_status '
+                '--formatter json'.format(nodename=nodename,
+                                          longnames=longnames))
+
+    def _update_node_status(self, services, statuses, config):
+        node_status, node_extra_info = self._get_rabbitmq_node_status(config)
+        services[RABBITMQ_SERVICE_KEY]['extra_info']['node_status'] = \
+            node_extra_info
+        statuses.append(node_status)
+
+    def _get_rabbitmq_node_status(self, config):
+        detailed_status = self._query_rabbitmq(config)
+        if not detailed_status:
+            return NodeServiceStatus.INACTIVE, {}
+
+        if 'error' in detailed_status:
+            return NodeServiceStatus.INACTIVE, detailed_status
+
+        return NodeServiceStatus.ACTIVE, detailed_status
+
+    @staticmethod
+    def _query_rabbitmq(config):
+        rabbitmq_cred = (config['username'], config['password'])
+        try:
+            response = get(RABBITMQ_URL, auth=rabbitmq_cred, verify=CA_PATH)
+        except RequestException as error:
+            logger.error(
+                'Failed getting RabbitMQ node status due to {0}'.format(error))
+            return None
+
+        return response.json()
+
+    @staticmethod
+    def _rabbitmq_collect_status_failed(services):
+        services['node_status'] = {}
+        services['cluster_status'] = {}
```

Comment:
I think both `node_status` and `cluster_status` should be under the `RabbitMQ` key?